### PR TITLE
Hold coachmark duration and position

### PR DESCRIPTION
**Pull request by gwil**
> longer duration on hold coachmark, don’t show below link unless it’s super close

Adjustments for hold coachmark timings and distance necessary from top of window.
Duration is longer when triggered by scroll, shorter when triggered from user interaction.

Demo: https://dist-lrullhwwjd.now.sh

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -67,12 +67,12 @@ export default function Vidy(opts) {
 	let toFlush, queue=[]; // batch-events queue & timer
 	let $ = this;
 
-	function toCoachTimer(vlink, ev, ms) {
+	function toCoachTimer(vlink, ev, delay, duration = 2e3) {
```
Comment (lukeed) [inline src/index.js]:
```suggestion
	function toCoachTimer(vlink, ev, delay, duration) {
```

File: src/index.js [hunk 2]
```diff
@@ -67,12 +67,12 @@ export default function Vidy(opts) {
...
 		clearTimeout(autoCoachmark);
 		autoCoachmark = setTimeout(() => {
-			dom.append(vlink, elements.holdCoachmark(vlink, ev));
+			dom.append(vlink, elements.holdCoachmark(vlink, ev, duration));
```
Comment (lukeed) [inline src/index.js]:
```suggestion
			dom.append(vlink, elements.holdCoachmark(vlink, ev, duration || 2e3));
```

**Comment by lukeed** [comment]
Why the switch to `clientX/Y`? They are different values with different reference points.

Not 100%, but I think this would break dashboard editor

**Comment by eudisd** [comment]
Good call.  Now that i think of it, I don't think I've tested in the dashboard.  Will do now and report back

**Comment by lukeed** [comment]
I can't remember which is which, but one will use the iframe's boundary as the reference point & the other will use the full window.

**Comment by eudisd** [comment]
Looking good on the dashboard in terms of xy coodinates.  The scroll-to-embed still works and things look good with the exeception of the vlink icon:

https://cl.ly/da2eae6bda27

**Comment by gwil** [comment]
> Why the switch to clientX/Y? They are different values with different reference points.

`pageY` was giving the position from the top of the page, `clientY` was giving the position from the top of the viewport. Needed to determine the distance to the top of the viewport to position the coachmark on top/bottom

**Pull request merged**